docs/Routes.md

```diff
@@ -168,7 +171,24 @@ fastify.get('/', options, async function (request, reply) {
   reply.send(processed)
 })
 ```
-**Warning:** If you use `return` and `reply.send` at the same time, the first one that happens takes precedence, the second value will be discarded, a *warn* log will also be emitted if the value is not `undefined`.
+* **Warning:** If you use `return` and `reply.send` at the same time, the first one that happens takes precedence, the second value will be discarded, a *warn* log will also be emitted because you tried to send a response twice.
+* **Warning:** You can't return `undefined`. For more details read [promise-resolution](#promise-resolution).
+
+<a name="promise-resolution"></a>
+### Promise resolution
+
+If you use `async/await` or return a promise inside your handler you should be aware of a special behaviour which is  necessary to support the callback and promise control-flow. You can't resolve a promise with `undefined` it is ignored, a *error* log will also be emitted because you pass the control of the response to the promise which can never be resolved. There are only two simple rules to remember:
```

Comment:
> You can't resolve a promise with `undefined` it is ignored, a *error* log will also be emitted because you pass the control of the response to the promise which can never be resolved.

If you resolve the handler's promise with `undefined`, it will be ignored causing the request to hang and an error log to be emitted.

___

What about this?

Reply:
It's all the same to me 😄

Reply:
Please always mention what you don't understand because the text can be written in many ways.

Reply:
I prefer a shorter version of this explanation, the user does not need to know *"because you pass the control of the response to the promise which can never be resolved."* IMHO.
I think we should explain what it will happen rather than describe the internals ;)

Reply:
Updated